io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -44,6 +38,163 @@ use crate::{
 
 use super::{BsError, Lvol, LvsError, LvsLvol};
 
+/// Result for low-level Lvol calls.
+pub type LvolResult = Result<*mut spdk_lvol, Errno>;
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait::async_trait(?Send)]
+pub trait LvolSnapshotOps {
+    type Error;
+    type SnapshotIter;
+    type Lvol;
+    /// Create Snapshot Common API.
+    async fn create_snapshot(
+        &self,
+        snap_param: SnapshotParams,
+    ) -> Result<Lvol, Self::Error>;
+
+    /// Create a snapshot in Remote.
+    async fn create_snapshot_remote(
+        &self,
+        nvmf_req: &NvmfReq,
+        snapshot_params: SnapshotParams,
+    );
+    /// Destroy snapshot.
+    async fn destroy_snapshot(mut self) -> Result<(), Self::Error>;
+
+    /// List Snapshot details based on source UUID from which snapshot is
+    /// created.
+    fn list_snapshot_by_source_uuid(&self) -> Vec<VolumeSnapshotDescriptor>;
```

Comment:
Prefer using standard terminology 

```suggestion
    fn list_snapshot_by_origin(&self) -> Vec<VolumeSnapshotDescriptor>;
```

Reply:
I did not change this naming, already existing.